docs/development/AutomaticInstrumentation.md

```diff
@@ -64,11 +64,73 @@ public class ClientQueryIteratorsIntegrations
 ```
 
 > Note that both `OnMethodBegin` and `OnMethodEnd` are optional. If you only need one of the integration points, you can omit the others
+
+#### `OnMethodEnd` and `OnMethodBegin` parameters
  
 The first parameter passed to the method is the instance on which the method is called (for `static` methods, this parameter should be omitted), and should be a _generic parameter_ type.  
- 
+
 For parameters that are well-known types like `string`, `object`, or `Exception`, you can use the type directly in the `OnMethodBegin` or `OnMethodEnd` methods. For other types that can't be directly referenced, such as types in the target-library, you should use generic parameters. If you need to manipulate the generic parameters, for example to access values, use the duck-typing approach described below.
 
+
+
+##### `OnMethodBegin`
+
+OnMethodBegin signatures with 1 or more parameters with 1 or more generics:
+```csharp
+      CallTargetState OnMethodBegin<TTarget>(TTarget instance);
+      CallTargetState OnMethodBegin<TTarget, TArg1>(TTarget instance, ref TArg1 arg1);
+      CallTargetState OnMethodBegin<TTarget, TArg1, TArg2>(TTarget instance, ref TArg1 arg1, ref TArg2);
+      CallTargetState OnMethodBegin<TTarget, TArg1, TArg2, ...>(TTarget instance, ref TArg1 arg1, ref TArg2, ...);
+      CallTargetState OnMethodBegin<TTarget>();
+      CallTargetState OnMethodBegin<TTarget, TArg1>(ref TArg1 arg1);
+      CallTargetState OnMethodBegin<TTarget, TArg1, TArg2>(ref TArg1 arg1, ref TArg2);
+      CallTargetState OnMethodBegin<TTarget, TArg1, TArg2, ...>(ref TArg1 arg1, ref TArg2, ...)
```

Comment:
These are only for `static` classes I think? Is it worth calling that out here (and in the `OnMethodEnd` signatures.

cc @tonyredondo - am I right about that?

Reply:
You’re right! Those are for static classes

Reply:
Added precision in [df2ec71](https://github.com/DataDog/dd-trace-dotnet/pull/2441/commits/df2ec71513a92cce721eb7f7100ffe0ac6c64fe7)